resources/docs/TUTORIAL_12_CLUSTERING.md

```diff
@@ -0,0 +1,122 @@
+Text Clustering in flair
+----------
+
+In this package text clustering is implemented. This module has the following
+clustering algorithms implemented:
+- k-Means
+- BIRCH
+- Expectation Maximization
+
+Each of the implemented algorithm needs to have an instanced DocumentEmbedding. This embedding will 
+transform each text/document to a vector. With these vectors the clustering algorithm can be performed.
+
+---------------------------
+
+k-Means
+------
+k-Means is a classical and well known clustering algorithm. k-Means is a partitioning-based Clustering algorithm. 
+The user defines with the parameter *k* how many clusters the given data has. 
+So the choice of *k* is very important. 
+More about k-Means can be read on the official [sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html).
+
+
+```
+from flair.models import ClusteringModel
+from flair.datasets import TREC_6
+from flair.embeddings import SentenceTransformerDocumentEmbeddings
+from sklearn.cluster import KMeans
+
+embeddings = SentenceTransformerDocumentEmbeddings()
+
+corpus = TREC_6().downsample(0.05)
```

Comment:
Add a remark the it is required to set memory_mode to full because otherwise we don't store all embeddings in memory which is required to perform clustering